cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -2171,6 +2171,24 @@ class ArrowDeserializer {
     return Status::OK();
   }
 
+  Status Visit(const ExtensionType& type) {
```

Comment:
An alternative to actually handling it here in the C++ side is to handle it on the cython/python side to ensure the storage array is passed to the ConvertArrayToPandas function

Reply:
It sounds ok to handle it on the C++ side, as for other types.